tensorflow/core/graph/mkl_optimizer_merge.cc

```diff
@@ -117,18 +261,9 @@ Node* NodeMergeRewritePass::FindNodeForMerge(const Node* a) const {
   return nullptr;
 }
 
-
-/*
-  Merge predecessor node with its successor.
-  Currently, we support merging Conv2D with AddBias, and
-  MatMul with Add node.
-
-  Returns true, if merging is successful and supported.
-  Returns false otherwise.
-*/
 bool NodeMergeRewritePass::MergeNode(std::unique_ptr<Graph>* g,
```

Comment:
Can you change these functions to return a Status instead of a bool?

Then, you should change the TF_CHECK_OK calls below (and elsewhere) to return an error Status instead of doing a check fail.

Also, you should change any checks like the one below where you check "data_format_pred != data_format_succ" to return a descriptive error message, rather than just false.